agent/api/ecsclient/client.go

```diff
@@ -226,7 +235,24 @@ func (client *APIECSClient) setInstanceIdentity(registerRequest ecs.RegisterCont
 	}
 
 	iidRetrieved := true
-	instanceIdentityDoc, err := client.ec2metadata.GetDynamicData(ec2.InstanceIdentityDocumentResource)
+	backoff := retry.NewExponentialBackoff(setInstIdRetryBackoffMin, setInstIdRetryBackoffMax,
+		setInstIdRetryBackoffJitter, setInstIdRetryBackoffMultiple)
+	ctx, cancel := context.WithTimeout(context.Background(), setInstIdRetryTimeOut)
+	defer cancel()
+	err := retry.RetryWithBackoffCtx(ctx, backoff, func() error {
+		var attemptErr error
+		seelog.Debugf("Attempting to get IID")
+		instanceIdentityDoc, attemptErr = client.ec2metadata.GetDynamicData(ec2.InstanceIdentityDocumentResource)
+		if attemptErr != nil {
+			seelog.Debugf("Unable to get instance identity document, retrying: %v", attemptErr)
+			// force credentials to expire in case they are stale but not expired (e.g. live update)
+			client.credentialProvider.Expire()
+			client.credentialProvider = instancecreds.GetCredentials(client.config.External.Enabled())
+			return apierrors.NewRetriableError(apierrors.NewRetriable(true), attemptErr)
+		}
+		seelog.Debugf("Successfully retrieved the IID")
```

Comment:
same here, can we change this to "Successfully retrieved the instance identity document"